sql/catalyst/src/test/scala/org/apache/spark/sql/connector/InMemoryTableCatalog.scala

```diff
@@ -181,9 +180,8 @@ class InMemoryTableCatalog extends BasicInMemoryTableCatalog with SupportsNamesp
   }
 
   override def dropNamespace(namespace: Array[String]): Boolean = {
-    if (listTables(namespace).nonEmpty) {
-      throw new IllegalStateException(s"Cannot delete non-empty namespace: ${namespace.quoted}")
-    }
+    listNamespaces(namespace).map(dropNamespace)
+    listTables(namespace).map(dropTable)
```

Comment:
We should use `foreach` in such cases to avoid potential issues (see https://issues.apache.org/jira/browse/SPARK-16694)

Reply:
Let me make a small followup although It is currently rather a nit since it doesn't affect the return type.

Reply:
Good catch!

To save time for others, the problem is that `map` is not guaranteed to be immediately evaluated, so this pattern could result in a noop.

Reply:
Ah, yeah. Thanks. Like `iteroatr.map()` -> iterator :-).

```scala
scala> Iterator(1,2,3).map(println)
res0: Iterator[Unit] = <iterator>

scala> Iterator(1,2,3).foreach(println)
1
2
3
```

worse:

```scala
scala> Iterator(1,2,3).map(println).toSeq
1
res15: Seq[Unit] = Stream((), ?)
```

Reply:
Ah, yeah. Thanks. Like `iteroatr.map()` -> iterator :-).

```scala
scala> Iterator(1,2,3).map(println)
res0: Iterator[Unit] = <iterator>

scala> Iterator(1,2,3).foreach(println)
1
2
3
```

worse:

```scala
scala> Iterator(1,2,3).map(println).toSeq
1
res1: Seq[Unit] = Stream((), ?)
```